birdie/src/routes/source/submit/+page.svelte

```diff
@@ -96,15 +100,46 @@
 			await emit('error', e);
 		}
 
-		submitting = false;
+		showProgressModal = false;
+		progressModalTitle = '';
+		loading = false;
+	};
+
+	const refreshLocks = async () => {
+		loading = true;
+		try {
+			repoStatus.set(await getRepoStatus());
+		} catch (e) {
+			await emit('error', e);
+		}
+		loading = false;
+	};
+
+	const handleLockSelected = async () => {
+		loading = true;
+		showProgressModal = true;
+		progressModalTitle = 'Locking Files';
+
+		try {
+			const selectedPaths = $selectedFiles.map((file) => file.path);
+			await lockFiles(selectedPaths);
+			await emit('success', 'Files locked!');
+			await verifyLocks();
+			await refreshLocks();
+		} catch (e) {
+			await emit('error', e);
+		}
+
+		showProgressModal = false;
+		progressModalTitle = '';
 		loading = false;
 	};
 
 	onMount(() => {
 		void refreshFiles(true);
 
 		const interval = setInterval(async () => {
-			if (!submitting) {
+			if (progressModalTitle === 'Submitting') {
```

Comment:
i think maybe just `if (!loading)` here maybe? checking a string value here gives me the ick